UM/Qt/Bindings/MainWindow.py

```diff
@@ -63,14 +63,22 @@ def __init__(self, parent = None):
                                   int(self._preferences.getValue("general/window_width")),
                                   int(self._preferences.getValue("general/window_height")))
 
-        if not self._preferences.getValue("general/restore_window_geometry"):
+        if (not self._preferences.getValue("general/restore_window_geometry") or
+                self._preferences.getValue("general/last_shutdown_severity") == Application.ShutdownSeverity.HardCrash):
             # Ignore whatever the preferences said.
-            Logger.log("i", "Not restoring window geometry from preferences because 'restore_window_geometry' is false")
+            Logger.log("i", "Window geometry not restored from preferences. "
+                            "See 'general/restore_window_geometry' and/or 'general/last_shutdown_severity' for cause.")
             restored_geometry = QRect(self.DEFAULT_WINDOW_LEFT,
                                       self.DEFAULT_WINDOW_TOP,
                                       self.DEFAULT_WINDOW_WIDTH,
                                       self.DEFAULT_WINDOW_HEIGHT)
```

Comment:
This is out of current subject, but why not use `QWidget::saveGeometry()` and `QWidget::restoreGeometry()` ? The stored content is not human-readable, but it's designed especially for this purpose.

Reply:
I think we want it to be human-editable as a last resort for our users. (It might also be easier for QA.)

Reply:
As long as you don't use a QMainWindow, that's good enough